app/services/check_answers/section_lister_service.rb

```diff
@@ -98,7 +98,11 @@ def benefits_fields(session_key: "benefits", field_type: "benefit")
     end
 
     def partner_benefits_fields
-      benefits_fields(session_key: "partner_benefits", field_type: "benefit")
+      if StepsHelper.valid_step?(@model, :partner_benefits)
```

Comment:
It looks to me like the buggy line is really line 84, in `#benefits_fields` which makes the assumption that the relevant step is `:benefits`, when in fact, when called from `#partner_benefits_fields`, the relevant step is `:partner_benefits`.

If we don't fix that buggy line then if there's ever a situation (I know, I know, it's unlikely) where `:benefits` is an _invalid_ step but `:partner_benefits` is a valid one, line 84 will bite us again, because we'll _not_ show the partner benefits section even when we should.

So I reckon the 'best' way to fix this bug is to fix the bad assumption in line 84, rather than try to work around it.

Reply:
Oh, sorry, I jumped the gun, I didn't see this PR was only a draft. Sorry!

Reply:
review appreciated anyway!